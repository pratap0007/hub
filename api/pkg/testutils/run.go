package testutils

import (
	"sync"
	"testing"
)

var setup sync.Once

func Setup(t *testing.T) *TestConfig {
	tc := Config()
	setup.Do(func() {
		if !tc.IsValid() {
			t.Fatalf("Failed to create test configuration object: %s", tc.Error())
		}

		if err := applyMigration(); err != nil {
			t.Fatalf("Failed to apply migration: %s", tc.Error())
		}
	})

	// NOTE: not calling tc.Cleanup as the connection
	//       need to be resued by other tests
	// t.Cleanup(func() {
	// 	 tc.Cleanup()
	// })

	return tc
}
