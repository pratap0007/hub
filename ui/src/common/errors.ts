export const apiDownError = `We are sorry but something went wrong`;
export const serverError = `Sorry, we're down for maintenance, please come back later`;
export const resourceNotFoundError = `No resources found`;
export const catalogConfigureError = `No resources found, Please configure the catalog!`;
