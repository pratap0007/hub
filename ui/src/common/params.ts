export enum Params {
  Query = 'query',
  SortBy = 'sortBy',
  Category = 'category',
  Kind = 'kind',
  Catalog = 'catalog',
  Platform = 'platform',
  Tag = 'tag',
  Code = 'code',
  Status = 'status'
}
