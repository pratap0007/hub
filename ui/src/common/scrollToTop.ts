import { assert } from '../store/utils';

// This function moves the page to top
export const scrollToTop = () => {
  const scroller = document.querySelector('main');
  assert(scroller);
  if (scroller) scroller.scrollTo(0, 0);
};
