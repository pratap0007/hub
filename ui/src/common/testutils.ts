//Sets a Fake date for test
export const FakeDate = () => {
  beforeEach(() => {
    global.Date.now = jest.fn(() => new Date('2021-01-01T10:20:30Z').getTime());
  });
};

//Sets an Actual date for test
export const ActualDate = () => {
  afterEach(() => {
    global.Date = Date;
  });
};
