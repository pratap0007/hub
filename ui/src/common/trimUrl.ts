/**
 * This trims the `/` character from url if it has as a last character
 */
const trimChar = `/`;

export const TrimUrl = (url: string) => {
  if (url === '') {
    return url;
  } else {
    return url.charAt(url.length - 1) !== trimChar ? url : url.slice(0, -1);
  }
};
