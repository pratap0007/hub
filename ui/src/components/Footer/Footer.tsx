import React from 'react';
import { Card, Grid, Text, GridItem, TextVariants, TextContent } from '@patternfly/react-core';
import '@patternfly/react-core/dist/styles/base.css';
import tekton from '../../assets/logo/tekton.png';
import './Footer.css';

const Footer: React.FC = () => {
  return (
    <React.Fragment>
      <Card className="footer-card">
        <Grid>
          <GridItem span={12} className="text">
            <a href="https://cd.foundation">
              <img src="https://tekton.dev/partner-logos/cdf.png" alt="tekton.dev" />
            </a>
          </GridItem>
          <GridItem span={12} className="text">
            <TextContent className="text-color">
              <Text component={TextVariants.h3}>
                Tekton is a{' '}
                <Text component={TextVariants.a} href="https://cd.foundation">
                  Continuous Delivery Foundation
                </Text>{' '}
                project.
              </Text>
            </TextContent>
          </GridItem>
          <GridItem span={12} className="logo-margin">
            <img src={tekton} alt="Tekton" className="logo-size" />
          </GridItem>
          <GridItem span={2}></GridItem>
          <GridItem span={8} className="text-color">
            <Text>
              © 2020 The Linux Foundation®. All rights reserved. The Linux Foundation has registered
              trademarks and uses trademarks. For a list of trademarks of The Linux Foundation,
              please see our{' '}
              <Text
                component={TextVariants.a}
                href="https://www.linuxfoundation.org/trademark-usage/"
              >
                Trademark Usage page
              </Text>
              . Linux is a registered trademark of Linus Torvalds.{' '}
              <Text component={TextVariants.a} href="https://www.linuxfoundation.org/privacy/">
                Privacy Policy
              </Text>{' '}
              and{' '}
              <Text component={TextVariants.a} href="https://www.linuxfoundation.org/terms/">
                Terms of Use
              </Text>{' '}
              .
            </Text>
          </GridItem>
          <GridItem span={2}></GridItem>
        </Grid>
      </Card>
    </React.Fragment>
  );
};

export default Footer;
