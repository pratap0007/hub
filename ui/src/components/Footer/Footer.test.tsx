import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '.';
import { FakeDate, ActualDate } from '../../common/testutils';

// Assign a dummy date to global Date inorder to avoid snapshaot updation
FakeDate();

// Assign current Date once test is over
ActualDate();

it('should render the footer component', () => {
  const component = renderer.create(<Footer />).toJSON();
  expect(component).toMatchSnapshot();
});
