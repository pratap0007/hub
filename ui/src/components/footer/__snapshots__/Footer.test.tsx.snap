// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe={true}
>
  <div
    className="pf-l-grid"
  >
    <div
      className="pf-l-grid__item pf-m-4-col"
    />
    <div
      className="pf-l-grid__item pf-m-4-col pf-m-12-row"
    >
      <div
        className="pf-l-flex"
      >
        <div
          className=""
        >
          <a
            href="https://cd.foundation"
          >
            <img
              alt="tekton.dev"
              src="https://tekton.dev/partner-logos/cdf.png"
            />
          </a>
        </div>
      </div>
      <div
        className="pf-l-flex"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <div
          className="pf-c-content"
        >
          <h1
            className=""
            data-pf-content={true}
            style={
              Object {
                "color": "white",
              }
            }
          >
            Tekton is a
             
            <a
              className=""
              data-pf-content={true}
              href="https://cd.foundation"
            >
              Continuous Delivery Foundation
            </a>
             
            project.
          </h1>
        </div>
      </div>
      <div
        className="pf-l-flex"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <div
          className="logo-size"
        >
          <img
            alt="Tekton"
            src="tekton.png"
          />
        </div>
      </div>
      <div
        className="pf-l-flex"
      >
        <div
          className="pf-c-card__footer"
        >
          <p
            className=""
            data-pf-content={true}
            style={
              Object {
                "color": "white",
                "textAlign": "center",
              }
            }
          >
            © 2020 The Linux Foundation®. All rights reserved. The Linux Foundation has registered trademarks and uses trademarks. For a list of trademarks of The Linux Foundation, please see our
             
            <a
              className=""
              data-pf-content={true}
              href="https://www.linuxfoundation.org/trademark-usage/"
            >
              Trademark Usage page
            </a>
            . Linux is a registered trademark of Linus Torvalds.
             
            <a
              className=""
              data-pf-content={true}
              href="https://www.linuxfoundation.org/privacy/"
            >
              Privacy Policy
            </a>
             
            and
             
            <a
              className=""
              data-pf-content={true}
              href="https://www.linuxfoundation.org/terms/"
            >
              Terms of Use
            </a>
             
            .
          </p>
        </div>
      </div>
    </div>
    <div
      className="pf-l-grid__item pf-m-4-col"
    />
  </div>
</article>
`;
